plugin-testing-framework/script/test.ts

```diff
@@ -0,0 +1,470 @@
+        testMessageResponse = undefined;
+    }
+
+    /**
+     * Sets up tests for each platform.
+     * Creates the test project directory and the test update directory.
+     * Starts required emulators.
+     */
+    function setupTests(): void {
+        it("sets up tests correctly", (done) => {
+            var promises: Q.Promise<string>[] = [];
+            
+            targetPlatforms.forEach(platform => {
+                promises.push(platform.getEmulatorManager().bootEmulator(restartEmulators));
+            });
+            
+            console.log("Building test project.");
+            // create the test project
+            promises.push(createTestProject(testRunDirectory)
+                .then(() => {
+                    console.log("Building update project.");
+                    // create the update project
+                    return createTestProject(updatesDirectory);
+                }));
+                
+            Q.all<string>(promises).then(() => { done(); }, (error) => { done(error); });
+        });
+    }
+
+    /**
+     * Creates a test project directory at the given path.
+     */
+    function createTestProject(directory: string): Q.Promise<string> {
+        return projectManager.setupProject(directory, templatePath, TestAppName, TestNamespace);
+    }
+
+    /**
+     * Creates and runs the tests from the projectManager and TestBuilderDescribe objects passed to initializeTests.
+     */
+    function createAndRunTests(targetPlatform: platform.IPlatform): void {
+        describe("CodePush", function() {
+            before(() => {
+                setupServer(targetPlatform);
+                return targetPlatform.getEmulatorManager().uninstallApplication(TestNamespace)
+                    .then(projectManager.preparePlatform.bind(projectManager, testRunDirectory, targetPlatform))
+                    .then(projectManager.preparePlatform.bind(projectManager, updatesDirectory, targetPlatform));
+            });
+            
+            after(() => {
+                cleanupServer();
+                return projectManager.cleanupAfterPlatform(testRunDirectory, targetPlatform).then(projectManager.cleanupAfterPlatform.bind(projectManager, updatesDirectory, targetPlatform));
+            });
+            
+            // Build the tests through the root test.
+            rootTest.create(onlyRunCoreTests, projectManager, targetPlatform);
+        });
+    }
+
+    // BEGIN TESTING //
+
+    describe("CodePush " + projectManager.getPluginName() + " Plugin", function () {
+        this.timeout(100 * 60 * 1000);
+        
+        if (shouldSetup) describe("Setting Up For Tests", () => setupTests());
+        else {
+            targetPlatforms.forEach(platform => {
+                var prefix: string = (onlyRunCoreTests ? "Core Tests " : "Tests ") + thisPluginPath + " on ";
+                describe(prefix + platform.getName(), () => createAndRunTests(platform));
+            });
+        }
+    });
+}
```

Comment:
i tried to put all the stuff someone using the plugin would want in here (excluding ProjectManager and Platforms, which need to be extended from their classes), but i guess i agree with u that this stuff can be moved around, especially given that now that i think about it, "PluginTestingFramework" is both a redundant and excessively long name for an import